File: frontend/src/service/recipe.js
import api from "../hook/api"

export const get_recipe_candidate = async (categoryId) => {
    const response = await api.post("/recipe/candidate", { category_id: categoryId })
    return response.data
}

export const generate_recipe = async ({ recipe_ref, spice_level, doneness }) => {
    const response = await api.post("/recipe/generate", { recipe_ref, spice_level, doneness })
    return response.data
}

File: backend/router/recipe.py
from fastapi import APIRouter, Depends
from database import get_db
from sqlalchemy.orm import Session
from service.recipe import RecipeService
from schema.recipe import CandidateRequest, CandidateResponse, GenerateRequest, GenerateResponse

router = APIRouter(prefix="/recipe")


@router.post("/candidate", response_model=CandidateResponse)
def get_recipe_candidate(req: CandidateRequest, db: Session = Depends(get_db)):
    return RecipeService.get_candidate(req.category_id, db)


@router.post("/generate", response_model=GenerateResponse)
def generate_recipe(req: GenerateRequest):
    return RecipeService.generate(req)
